integration-tests/cypress/cypress.spec.js

```diff
@@ -40,272 +40,117 @@ function getCypressVersion () {
-        )
-        assert.equal(passedTest.content.meta[TEST_STATUS], 'pass')
-        assert.equal(failedTest.content.meta[TEST_STATUS], 'fail')
-        assert.include(failedTest.content.meta[ERROR_MESSAGE], 'error in after each hook')
-        assert.equal(skippedTest.content.meta[TEST_STATUS], 'skip')
-        assert.equal(testHookSuite.content.meta[TEST_STATUS], 'fail')
-        assert.include(testHookSuite.content.meta[ERROR_MESSAGE], 'error in after each hook')
-
-        // describe level hooks
-        const describeHookSuite = events.find(
-          event => event.content.resource === 'test_suite.cypress/e2e/hook-describe-error.cy.js'
-        )
-        const passedTestDescribe = events.find(
-          event => event.content.resource === 'cypress/e2e/hook-describe-error.cy.js.after passes'
-        )
-        const failedTestDescribe = events.find(
-          event => event.content.resource === 'cypress/e2e/hook-describe-error.cy.js.after will be marked as failed'
-        )
-        const skippedTestDescribe = events.find(
-          event => event.content.resource === 'cypress/e2e/hook-describe-error.cy.js.before will be skipped'
-        )
-        assert.equal(passedTestDescribe.content.meta[TEST_STATUS], 'pass')
-        assert.equal(failedTestDescribe.content.meta[TEST_STATUS], 'fail')
-        assert.include(failedTestDescribe.content.meta[ERROR_MESSAGE], 'error in after hook')
-        assert.equal(skippedTestDescribe.content.meta[TEST_STATUS], 'skip')
-        assert.equal(describeHookSuite.content.meta[TEST_STATUS], 'fail')
-        assert.include(describeHookSuite.content.meta[ERROR_MESSAGE], 'error in after hook')
-      }, 25000)
-
-    const {
-      NODE_OPTIONS, // NODE_OPTIONS dd-trace config does not work with cypress
-      ...restEnvVars
-    } = getCiVisEvpProxyConfig(receiver.port)
-
-    childProcess = exec(
-      `./node_modules/.bin/cypress run --quiet ${commandSuffix}`,
-      {
-        cwd,
-        env: {
-          ...restEnvVars,
-          CYPRESS_BASE_URL: `http://localhost:${webAppPort}`
-        },
-        stdio: 'pipe'
-      }
-    )
-    childProcess.on('exit', () => {
-      receiverPromise.then(() => {
-        done()
-      }).catch(done)
+const hookFile = 'dd-trace/loader-hook.mjs'
+
+const moduleType = [
+  {
+    type: 'commonJS',
+    testCommand: function commandWithSuffic (version) {
+      const commandSuffix = version === '6.7.0' ? '--config-file cypress-config.json' : ''
+      return `./node_modules/.bin/cypress run --quiet ${commandSuffix}`
+    }
+  },
+  {
+    type: 'esm',
+    testCommand: `node --loader=${hookFile} ./cypress-esm-config.mjs`
+  }
+]
+
+moduleType.forEach(({
+  type,
+  testCommand
+}) => {
+  describe(`${type} cypress@${version}`, function () {
+    // cypress only supports esm on versions >= 10.0.0
+    if (type === 'esm' && version < '10.0.0') {
```

Comment:
do we need `semver` here as well?

Reply:
it seems so: 6.7.0 is being tested with esm here: https://github.com/DataDog/dd-trace-js/actions/runs/5919271282/job/16049382951?pr=3555. It seems cucumber was working out of luck